test/0/0.transaction.unit.spec.js

```diff
@@ -77,6 +80,62 @@ describe('Transaction Unit', function () {
         assert.isObject(tx.getMetadata());
         assert.strictEqual(JSON.stringify(tx.getMetadata()), JSON.stringify(meta));
 
+        //** Test External Signing
+        
+        //test signing with private key externally, this will simulate an external signature such as from the Ledger
+        let sk = fctAddrUtils.addressToKey("Fs1PkAEbmo1XNangSnxmKqi1PN5sVDbQ6zsnXCsMUejT66WaDgkm");
+        let key = nacl.keyPair.fromSeed(sk);
+ 
+        tx = new TransactionBuilder(testTokenChainId)
+            .input(key.publicKey, 150)
+            .output("FA3aECpw3gEZ7CMQvRNxEtKBGKAos3922oqYLcHQ9NqXHudC6YBM", 150)
+            .build()
+            
+        //this should throw error for adding input to transaction error, when expecting signatures only
+        assert.throws(() => new TransactionBuilder(tx)
+            .input(key.publicKey, 150)
+            .pkSignature(key.publicKey, "abcdef0123456789")
+            .build())
+            
+        //this should throw error for having a publicKey that doesn't match input
+        assert.throws(() => new TransactionBuilder(tx)
+            .pkSignature("badc0de", "abcdef0123456789")
```

Comment:
`badc0de` should be replace with a valid public key that doesn't match for a stronger test case. This method also doesn't check whether the key `publicKey` is valid

Reply:
replaced with an public key generated from a different private key than the default one used.